tensordict/tensordict.py

```diff
@@ -1620,6 +1620,12 @@ def flatten_keys(
         for key, meta_value in self.items_meta():
             if meta_value.is_tensordict():
                 to_flatten.append(key)
+            elif separator in key and isinstance(
+                _exists(self, key.split(separator)), (MemmapTensor, Tensor)
```

Comment:
I don't think we need exist 
We could just ask if key is in TensorDict.keys(True) which will return all the keys, including nested ones.

Reply:
Working on this...thanks for pointing out the utility for returning nested keys.

The test to make sure below works with `flatten_keys()` is failing with the new approach:

```python
td5 = TensorDict(
        {f"a{separator}b": torch.zeros(3), "a": {"b": {"c": torch.zeros(3)}}}, []
     )
```

Reply:
Can you push this with the failing test?

Reply:
Pushed. Here's what's happening:
```python
breaking key: a, b
breaking key split: ('a', 'b')
all keys (including nested): ['a-b', ('a', 'b', 'c'), ('a', 'b'), 'a']
```
The recursive fashion in which the nested keys are included causes the flattened key to match and throws and error for this situation.

Reply:
I tried something, let me know if you think it's appropriate

Reply:
Looks good, thanks